src/doxygen.cpp

```diff
@@ -725,7 +725,8 @@ static Definition *buildScopeFromQualifiedName(const QCString &name_,SrcLangExt
       if (newNd)
       {
         newNd->setLanguage(lang);
-        newNd->setArtificial(TRUE);
+        // looks like an atrificial namespace, but is in fact part of ns1.ns2, issue 10953
+        newNd->setArtificial(false);
```

Comment:
@albert-github I fear this change will have negative side effects, such as also creating empty pages. In general, the class is 'artificial' in the sense that is it introduced by Doxygen, without finding the definition in the code. Maybe for Java packages we need to treat things differently...

Reply:
I could not think of any side effects but is well possible.
I remember that there have been some issues with CSharp as well regarding nesting. might be (at first glance in namespacedef.cpp or (more likely) in scanner.l. In the later I see with the rule:
```
<Using>({ID}{BN}*("::"|"."){BN}*)*({ID}|{OPERATOR})
```
the comment:
```
                                          if (yyextra->insideCS) /* Hack: in C# a using declaration and
                                                                    directive have the same syntax, so we
                                                                    also add it as a using directive here
                                                                  */
```

it should be checked if this is also possible for Java packages, probably, with the rule (limited to the case that we are in Java):
```
<PackageName>{ID}(("."|"\\"){ID})*
```

Reply:
I have made a different fix, see https://github.com/doxygen/doxygen/commit/269d60d6057f67b587e491ef5b1336fd071b96a8. 
See problem also happened for C++. e.g.
```c++
namespace X
{
  /** namespace Y */
  namespace Y
  {
    /** YY */
    class YY
    {
    };
  }
}
```

Reply:
Looks OK to me.